src/Client.ts

```diff
@@ -177,54 +189,136 @@ export class Client extends ClientJS {
           this.decorators.trigger(on.event, this, true)
         );
       } else {
-        this.on(
-          on.event as any,
-          this.decorators.trigger(on.event, this)
-        );
+        this.on(on.event as any, this.decorators.trigger(on.event, this));
       }
     });
 
-    return await super.login(token);
+    return super.login(token);
   }
 
   /**
    * Initialize all the @Slash with their permissions
    */
   async initSlashes() {
-    await Promise.all(
-      this.slashes.map(async (slash) => {
-        // Init all the @Slash
-        if (slash.guilds.length > 0) {
-          // If the @Slash is guild specific, add it to the guild
-          await Promise.all(
-            slash.guilds.map(async (guildID) => {
-              const guild = this.guilds.cache.get(guildID as Snowflake);
+    // # first init guild's all slash
+    const guildSlashStorage = new Map<string, DSlash[]>();
+    const guildsSlash = this.slashes.filter((s) => s.guilds?.length);
+
+    // group single guild slash together
+    for (const s of guildsSlash) {
+      s.guilds.forEach((guild) =>
+        guildSlashStorage.set(guild, [
+          ...(guildSlashStorage.get(guild) ?? []),
+          s,
+        ])
+      );
+    }
 
-              if (!guild) {
-                throw new GuildNotFoundError(guildID);
-              }
+    // update guild commands
+    for (const gc of guildSlashStorage) {
+      const guild = await this.guilds.fetch({ guild: gc[0] as Snowflake });
```

Comment:
This could be slow if there are many guilds - consider replacing for-loop with `Promise.all(guildSlashStorage.map(...))`

Reply:
the guildSlashStorage is already a map, can not call it. please suggest other way.